ls1intum/Artemis
# `Iris`: Replace or remove unnecessary client dependencies

### Checklist
#### General


#### Client

### Motivation and Context



The IRIS UI is currently using these dependencies:
- `angular2-draggable`: Used to move and resize the chat window. Can be replaced with already used `interact.js`.
- `angular-animations`: Provides a predefined animation. We don't need that at all
- `css-element-queries`: Used to monitor the resizing of the chat window and storing the values in local storage. We don't need that at all

### Description


Remove or replace dependencies as described above.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Programming Exercise with IRIS enabled on TS9

1. Go to the exercise view
2. Open the chat window
3. Verify that you can drag the window around
4. Verify that you can resize the window on all edges
5. Verify that it expands to 95vw and 85vh if you click the fullscreen icon
6. Verify that it reverts back to default if you click it again
7. Verify that the icon automatically changes if you manually drag the window larger than 95vm and 85vh

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage







### Screenshots
